src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs

```diff
@@ -84,6 +85,12 @@ public override async Task RunAsync(CancellationToken cancellationToken)
             foreach (var (name, secret, secretType, secretReferences) in orderedSecretTypes)
             {
                 _console.WriteLine($"Synchronizing secret {name}, type {secret.Type}");
+
+                if (!existingSecrets[name].Tags.ContainsKey(AzureKeyVault.NextRotationOnTag))
```

Comment:
Just asking as I don't know the code much:
What if the secret is not in the KeyVault? Is it possible? Will this throw?

Reply:
I rewrote that part so it doesn't throw.
Later in the code, we go through all secrets that are in the manifest but not in the Vault and create them, so we shouldn't be doing anything here if we don't find the secret